src/main/java/org/terasology/climateConditions/HyperthermiaSystem.java

```diff
@@ -124,12 +69,44 @@ private void applyWeakening(EntityRef player, HealthComponent health, Hypertherm
 
 
     /**
-     *  Reverts the player weakening by restoring the maxHealth and regeneration of the player to the original value.
+     * Reverts the player weakening by restoring the maxHealth and regeneration of the player to the original value.
      */
     private void revertWeakening(EntityRef player, HealthComponent health, HyperthermiaComponent hyperthermia) {
         player.send(new ChangeMaxHealthEvent(player.getParentPrefab().getComponent(HealthComponent.class).maxHealth));
         player.send(new ActivateRegenEvent());
         health.regenRate /= hyperthermia.regenMultiplier;
         player.saveComponent(health);
     }
+
+    @ReceiveEvent
+    public void hyperthermiaLevelChanged(HyperthermiaLevelChangedEvent event, EntityRef player,
+                                         HyperthermiaComponent hyperthermia, HealthComponent health) {
+        modifyHyperthermiaMultipliers(player, hyperthermia, event.getNewLevel());
```

Comment:
The `player` is only passed to the method so that the component can be saved. It may be a bit cleaner to move that part out of the method:
```suggestion
        player.saveComponent(modifyHyperthermiaMultipliers(hyperthermia, event.getNewLevel()));
```
This change also requires the method to return a `HyperthermiaComponent`, but achieving that is quite simple as we can just return the exact component we modify 😉